src/server/routes/editor.js

```diff
@@ -512,4 +512,111 @@ router.post('/:id/achievements/', auth.isAuthenticated, (req, res) => {
 	handler.sendPromiseResult(res, rankPromise);
 });
 
+
+async function getOrderedCollections(from, size, entityType, req) {
+	const {UserCollection} = req.app.locals.orm;
+	const isThisCurrentUser = req.user && parseInt(req.params.id, 10) === parseInt(req.user.id, 10);
+
+	const allCollections = await new UserCollection()
+		.query((qb) => {
+			qb.leftJoin('bookbrainz.user_collection_collaborator',
+				'bookbrainz.user_collection.id', '=',
+				'bookbrainz.user_collection_collaborator.collection_id');
+		})
+		.where((builder) => {
+			if (!isThisCurrentUser) {
+				builder.where('public', true);
+			}
+			if (entityType) {
+				builder.where('entity_type', entityType);
+			}
+		})
+		.where((builder) => {
+			builder.where('editor_id', '=', req.params.id).orWhere('owner_id', '=', req.params.id);
+		})
+		.fetchPage({
+			limit: size,
+			offset: from
+		});
+
+	const collectionsJSON = allCollections ? allCollections.toJSON() : [];
+
+	collectionsJSON.forEach((collection) => {
+		collection.isOwner = parseInt(req.params.id, 10) === collection.ownerId;
+	});
+
+	return collectionsJSON;
+}
+
+// eslint-disable-next-line consistent-return
+router.get('/:id/collections', async (req, res, next) => {
+	const {Editor, TitleUnlock} = req.app.locals.orm;
+	const size = req.query.size ? parseInt(req.query.size, 10) : 20;
+	const from = req.query.from ? parseInt(req.query.from, 10) : 0;
+	let type = req.query.type ? req.query.type : null;
+	const entityTypes = _.keys(utils.getEntityModels(req.app.locals.orm));
+	if (!entityTypes.includes(type)) {
+		type = null;
+	}
```

Comment:
this is required because `entity_type` in the table `User_Collection` can only take valid values. Otherwise it throws an error